zerver/views/custom_profile_fields.py

```diff
@@ -94,12 +109,40 @@ def validate_custom_profile_field(
     if field_type not in field_types:
         raise JsonableError(_("Invalid field type."))
 
+    validate_display_in_profile_summary_field(field_type, display_in_profile_summary)
+
 
 check_profile_field_data: Validator[ProfileFieldData] = check_dict(
     value_validator=check_union([check_dict(value_validator=check_string), check_string])
 )
 
 
+def update_only_display_in_profile_summary(
+    requested_name: str,
+    requested_hint: str,
+    requested_field_data: ProfileFieldData,
+    existing_field: CustomProfileField,
+) -> bool:
+    if (
+        requested_name != existing_field.name
+        or requested_hint != existing_field.hint
+        or requested_field_data != orjson.loads(existing_field.field_data)
+    ):
+        return False
+    return True
+
+
+def display_in_profile_summary_reach_limit(profile_field_id: int) -> bool:
+    # Excluding requested profile field while updating, because if its "display_in_profile_summary"
+    # value is true it can raise JsonableError for limit reach.
+    return (
+        CustomProfileField.objects.filter(display_in_profile_summary=True)
+        .exclude(id=profile_field_id)
+        .count()
+        >= max_display_in_profile_summary_fields
+    )
```

Comment:
I rewrote this to be a bit more readable, and be able to be reused rather than duplicated in `create_realm_custom_profile_field` .